.idea/checkstyle-idea.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CheckStyle-IDEA">
+    <option name="configuration">
+      <map>
+        <entry key="checkstyle-version" value="8.26" />
+        <entry key="copy-libs" value="true" />
+        <entry key="location-0" value="BUNDLED:(bundled):Sun Checks" />
+        <entry key="location-1" value="BUNDLED:(bundled):Google Checks" />
+        <entry key="scan-before-checkin" value="false" />
+        <entry key="scanscope" value="JavaOnly" />
+        <entry key="suppress-errors" value="false" />
+      </map>
+    </option>
+  </component>
+</project>
```

Comment:
Is a `.idea` folder really necessary? Normally that is an automatically created folder by Intellij and does not need to be part of the repo?

Also, missing newline at the end. This is missing in multiple files. It can be enabled in Intellij so that it automatically adds a newline if missing.

Reply:
https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
The idea folder is there to share configurations for different user if you don't use it the current setup would not work anymore